packages/auth-client/src/messages/verify.ts

```diff
@@ -68,37 +79,35 @@ const validateDomain = (message: SiweMessage, domain: string): AuthClientResult<
 };
 
 const verifySiweMessage = async (
-  message: SiweMessage,
-  signature: string,
-  provider?: Provider,
-): AuthClientAsyncResult<SiweResponse> => {
-  return ResultAsync.fromPromise(message.verify({ signature }, { provider, suppressExceptions: true }), (e) => {
+  message: string,
+  signature: Hex,
+  client: PublicClient<Transport, typeof optimism>,
+): AuthClientAsyncResult<boolean> => {
+  const parsed = parseSiweMessage(message);
+  return ResultAsync.fromPromise(client.verifySiweMessage({ signature, message, ...parsed }), (e) => {
     return new AuthClientError("unauthorized", e as Error);
   });
 };
 
 const verifyFidOwner = async (
-  response: SiweResponse & FarcasterResourceParams,
-  fidVerifier: (custody: Hex) => Promise<BigInt>,
-): AuthClientAsyncResult<SiweResponse & FarcasterResourceParams> => {
-  const signer = response.data.address as Hex;
+  message: SiweMessage & FarcasterResourceParams,
+  fidVerifier: (custody: Hex) => Promise<bigint>,
+): AuthClientAsyncResult<SiweMessage> => {
+  const signer = message.address as Hex;
   return ResultAsync.fromPromise(fidVerifier(signer), (e) => {
     return new AuthClientError("unavailable", e as Error);
   }).andThen((fid) => {
-    if (fid !== BigInt(response.fid)) {
-      response.success = false;
-      response.error = new SiweError(
-        `Invalid resource: signer ${signer} does not own fid ${response.fid}.`,
-        response.fid.toString(),
-        fid.toString(),
```

Comment:
this did not make much sense to me, so now it will throw an error as below.

previously it both threw an error during `.unwrap` and additionally returned `.error` and `.success` for that specific error.

I believe it's better to stick to one error handling flow. If this change is not appreciated I can make it up as before.

Reply:
Good change, it's cleaner. IIRC this was a hack around how we were catching and formatting `SiweError`. Now that we're not using it, not necessary.